### DISCUSSION TITLE
Log Batch Purger and scheduler

### DISCUSSION THREAD

**Discussion by nishanth-yb** [jongpie/NebulaLogger#724] [Q&A] [answered]
Hi Team,

I need some help regarding the Nebula Logger's purge functionality and would really appreciate your guidance.

1. When running the LogBatchPurger class, does it log the deletions of logs and log entries, thus consuming additional data storage?
2. Can we modify the LogBatchPurger class to retain logs and log entries from the current month and delete all entries older than that?

Thanks in advance for your guidance!

**Comment by jongpie** [comment, marked as answer]
Hi @nishanth-yb there's some info about this [in README](https://github.com/jongpie/NebulaLogger/tree/v4.13.15#batch-deleting-with-apex-jobs), but essentially:
- The date field `Log__c.LogRetentionDate__c` controls when a `Log__c` record is deleted (as well as its related `LogEntry__c` records).
  - `LogBatchPurger` will delete any records where `LogRetentionDate__c <= System.today()`
  - To keep a record, you can set `LogRetentionDate__c` to `null`
- You can use the custom settings field `LoggerSettings__c.NumberOfDaysToRetainLogs__c` to customize (including per profile or user if you want) the value saved in `Log__c.LogRetentionDate__c`.
  - By default, `LoggerSettings__c.NumberOfDaysToRetainLogs__c` is populated with "TODAY + 14 DAYS" - so by default, all `Log__c` records will be kept for 14 days before `LogBatchPurger` will delete it

Ultimately, you just need to set the value of `Log__c.LogRetentionDate__c` to control when data is deleted (using either `LoggerSettings__c`, or building your own automation to set it, or simply manually editing the field). 

In your case, if you want to keep logs for the current month, you could update the value of `LoggerSettings__c.NumberOfDaysToRetainLogs__c` from `14` to `30` (using 30 days as approximately 1 month) - any new `Log__c` records would then be kept for 30 days before being deleted.

Hope this helps!

**Comment by nishanth-yb** [comment]
Hey Jonathan,

Makes sense, thank you so much for the clarification.
This helps a lot.

Thanks again,
Nishanth
